osgp/protocol-adapter-oslp/web-device-simulator/src/main/java/org/opensmartgridplatform/webdevicesimulator/service/OslpChannelHandler.java

```diff
@@ -1200,8 +1117,14 @@ private void handleSetEventNotificationsRequest(final Device device, final SetEv
     private void handleUpdateFirmwareRequest(final Device device, final UpdateFirmwareRequest request) {
         LOGGER.debug("handle UpdateFirmwareRequest for device: {}, with serialized size of {}",
                 device.getDeviceIdentification(), request.getSerializedSize());
-        // For now, do nothing, perhaps store firmware version, so that it can
-        // be displayed ???
+
+        this.sendDelayedDeviceRegistration(device);
+
+        // Send a firmware activation event after the device registration has
+        // (likely) been finished
```

Comment:
The "likely" here looks a little suspicious. Could it be made somewhat safer if the delay sending the device registration were included as a method parameter of sendDelayedDeviceRegistration just above here?

Further, like before, if the delays are likely suitable for different environments it is OK as it is, but if we already think there is a realistic chance we would like to vary this between different uses, it might be an idea to make them configurable.

Reply:
Since the registration message implementation is using a 2-second delay, and the event sending implementation, which has been added as part of this pull request, is using a 7-second delay, it is very likely that everything works as expected. Because this is simulator code, not changing it for now or moving it to properties.